spring-cloud-gcp-vision/src/main/java/org/springframework/cloud/gcp/vision/CloudVisionTemplate.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.vision;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesRequest;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.protobuf.ByteString;
+
+import org.springframework.core.io.Resource;
+
+/**
+ * Spring Template offering convenience methods for interacting with the Cloud Vision
+ * APIs.
+ *
+ * @author Daniel Zou
+ *
+ * @since 1.1
+ */
+public class CloudVisionTemplate {
+
+	private final ImageAnnotatorClient imageAnnotatorClient;
+
+	public CloudVisionTemplate(ImageAnnotatorClient imageAnnotatorClient) {
+		this.imageAnnotatorClient = imageAnnotatorClient;
+	}
+
+	/**
+	 * Analyze an image and extract the feature of the image specified by {@code featureType}.
+	 *
+	 * <p>
```

Comment:
The `<p>` must be a first symbol in the paragraph. Only in method Javadocs: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting